common/src/main/java/com/sequenceiq/cloudbreak/quartz/statuschecker/service/StatusCheckerJobService.java

```diff
@@ -54,20 +56,21 @@
 
     public <T> void schedule(JobResourceAdapter<T> resource) {
         JobDetail jobDetail = buildJobDetail(resource);
-        Trigger trigger = buildJobTrigger(jobDetail, RANDOM.nextInt(statusCheckerConfig.getIntervalInSeconds()), statusCheckerConfig.getIntervalInSeconds());
-        schedule(jobDetail, trigger, resource.getLocalId());
+        Trigger trigger = buildJobTrigger(jobDetail, resource.getJobResource(), RANDOM.nextInt(statusCheckerConfig.getIntervalInSeconds()),
+                statusCheckerConfig.getIntervalInSeconds());
+        schedule(jobDetail, trigger, resource.getJobResource());
     }
 
     public <T> void schedule(JobResourceAdapter<T> resource, int delayInSeconds) {
         JobDetail jobDetail = buildJobDetail(resource);
-        Trigger trigger = buildJobTrigger(jobDetail, delayInSeconds, statusCheckerConfig.getIntervalInSeconds());
-        schedule(jobDetail, trigger, resource.getLocalId());
+        Trigger trigger = buildJobTrigger(jobDetail, resource.getJobResource(), delayInSeconds, statusCheckerConfig.getIntervalInSeconds());
+        schedule(jobDetail, trigger, resource.getJobResource());
     }
 
-    public void schedule(Long id, Class<? extends JobResourceAdapter<?>> resource) {
+    public void schedule(Long id, Class<? extends JobResourceAdapter<?>> resourceAdapterClass) {
         try {
-            Constructor<? extends JobResourceAdapter<?>> c = resource.getConstructor(Long.class, ApplicationContext.class);
-            JobResourceAdapter<?> resourceAdapter = c.newInstance(id, applicationContext);
+            Constructor<? extends JobResourceAdapter> c = resourceAdapterClass.getConstructor(Long.class, ApplicationContext.class);
```

Comment:
What is the benefit of passing the whole application context and id to the adapter class, instead of querying the resource here in the service, and passing that to the adapter?

Reply:
However, I just changed the parameter name here but I think in the JobResourceAdapter easier to inject the repository than in the scheduler module (more abstraction, interfaces and so on to inject and lookup the specific repo like StackRepository or EnvironmentRepository or DatalakeRepository).

Reply:
I would prefer it the other way, but it is not a show stopper